runtime/oti/VMHelpers.hpp

```diff
@@ -2081,25 +2092,35 @@ class VM_VMHelpers
 
 	/**
 	 * Check if the related J9VMContinuation is mounted to carrier thread
+	 * if carrierThreadID has been set in j9vmcontinuation->state, the continuation might be mounted,
+	 * there also is pending to be mounted case, when the mounting is blocked by the concurrent continuation
+	 * scanning or related vm access, so we need to confirm if carrierThread->currentContinuation
+	 * has been set for mounted case.
+	 *
 	 * @param[in] continuation the related J9VMContinuation
 	 * @return true if it is mounted.
 	 */
 	static VMINLINE bool
-	isContinuationMounted(J9VMContinuation *continuation)
+	isContinuationMountedFromContinuationState(uintptr_t continuationState)
 	{
-		return J9_ARE_ANY_BITS_SET(continuation->state, ~(uintptr_t)J9_GC_CONTINUATION_STATE_CONCURRENT_SCAN_ANY);
+		bool mounted = false;
+		J9VMThread *carrierThread = getCarrierThreadFromContinuationState(continuationState);
+		if (NULL != carrierThread && !isPendingToBeMountedFromContinuationState(continuationState)) {
+			mounted = true;
+		}
+		return mounted;
 	}
 
 	static VMINLINE J9VMThread *
 	getCarrierThreadFromContinuationState(uintptr_t continuationState)
 	{
-		return (J9VMThread *)(continuationState & (~(uintptr_t)J9_GC_CONTINUATION_STATE_CONCURRENT_SCAN_ANY));
+		return (J9VMThread *)(continuationState & J9_GC_CONTINUATION_STATE_CARRIERID_MASK);
 	}
 
 	static VMINLINE bool
 	isContinuationMountedOrConcurrentlyScanned(J9VMContinuation *continuation, bool isGlobalGC)
 	{
-		return isContinuationMounted(continuation) || isConcurrentlyScannedFromContinuationState(continuation->state, isGlobalGC);
+		return isContinuationMountedFromContinuationState(continuation->state) || isConcurrentlyScannedFromContinuationState(continuation->state, isGlobalGC);
```

Comment:
it would be wise to read the state in a local first, to avoid potential reading it twice and getting different values